File: api/UserManagement/UserManagement/Models/DTO/CreateUserDto.cs
namespace UserManagement.Models.DTO
{
    public class CreateUserDto
    {
        public required string FirstName { get; set; }

        public required string LastName { get; set; }

        public required string Email { get; set; }

        public string? Phone { get; set; }

        public required string Username { get; set; }

        public required string Password { get; set; }

        public Guid RoleId {  get; set; }

        public required List<CreateUserPermissionDto> Permissions { get; set; }
    }
}

File: app/user-management/src/app/models/create-user-dto.ts
export interface CreateUserDto {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  username: string;
  password: string;
  roleId: string;
  permissions: CreateUserPermissionDto[];
}
export interface CreateUserPermissionDto {
  permissionId: string;
  isReadable: boolean;
  isWritable: boolean;
  isDeletable: boolean;
}
